agent/config/config_unix.go

```diff
@@ -103,7 +103,7 @@ func DefaultConfig() Config {
 		CgroupCPUPeriod:                     defaultCgroupCPUPeriod,
 		GMSACapable:                         parseGMSACapability(),
 		GMSADomainlessCapable:               parseGMSADomainlessCapability(),
-		FSxWindowsFileServerCapable:         BooleanDefaultFalse{Value: ExplicitlyDisabled},
+		FSxWindowsFileServerCapable:         BooleanDefaultTrue{Value: ExplicitlyDisabled},
```

Comment:
since this feature is not supported on Linux, we would want to keep this as is right? 

`BooleanDefaultFalse{Value: ExplicitlyDisabled}`

Reply:
There is a universal contract of the type in `types.go`. We cannot change one without forking them out. I don't think it is a big enough case to fork this out. Therefore, keeping it as it.

Reply:
synced offline, this is because `FSxWindowsFileServerCapable` is a shared config type between both Windows, Linux platforms.

We've added tests that validate that this config type is true for Windows and false for everything else. Resolving related threads below too.